src/backends/pgfplotsx.jl

```diff
@@ -0,0 +1,844 @@
+               if is3d(series) || st == :heatmap
+                   series_func = PGFPlotsX.Plot3
+               else
+                   series_func = PGFPlotsX.Plot
+               end
+               if series[:fillrange] !== nothing && !isfilledcontour(series) && series[:ribbon] === nothing
+                   push!(series_opt, "area legend" => nothing)
+               end
+               if st == :heatmap
+                   push!(axis.options,
+                   "view" => "{0}{90}",
+                   "shader" => "flat corner",
+                   )
+               end
+               # treat segments
+               segments = if st in (:wireframe, :heatmap, :contour, :surface, :contour3d)
+                   iter_segments(surface_to_vecs(series[:x], series[:y], series[:z])...)
+               else
+                   iter_segments(series)
+               end
+               for (i, rng) in enumerate(segments)
+                   segment_opt = PGFPlotsX.Options()
+                   segment_opt = merge( segment_opt, pgfx_linestyle(opt, i) )
+                   if opt[:markershape] != :none
+                       marker = opt[:markershape]
+                       if marker isa Shape
+                           x = marker.x
+                           y = marker.y
+                           scale_factor = 0.025
+                           mark_size = opt[:markersize] * scale_factor
+                           path = join(["($(x[i] * mark_size), $(y[i] * mark_size))" for i in eachindex(x)], " -- ")
+                           c = get_markercolor(series, i)
+                           a = get_markeralpha(series, i)
+                           PGFPlotsX.push_preamble!(pgfx_plot.the_plot,
+                           """
+                           \\pgfdeclareplotmark{PlotsShape$(series_index)}{
+                           \\filldraw
+                           $path;
+                           }
+                           """
+                           )
+                       end
+                       segment_opt = merge( segment_opt, pgfx_marker(opt, i) )
+                   end
+                   if st == :shape ||
+                       isfilledcontour(series) ||
+                       series[:ribbon] !== nothing
+                       segment_opt = merge( segment_opt, pgfx_fillstyle(opt, i) )
+                   end
+                   # add fillrange
+                   if series[:fillrange] !== nothing && !isfilledcontour(series) && series[:ribbon] === nothing
+                       pgfx_fillrange_series!( axis, series, series_func, i, _cycle(series[:fillrange], rng), rng)
+                       # add to legend?
+                       if i == 1 && opt[:label] != "" && sp[:legend] != :none && should_add_to_legend(series)
+                           io = IOBuffer()
+                           PGFPlotsX.print_tex(io, pgfx_fillstyle(opt, i))
+                           style = strip(String(take!(io)),['[',']', ' '])
```

Comment:
Would it perhaps be nicer to have a function within PGFPlotsX.jl that returns the string rather than printing it? @KristofferC

Reply:
You can always get a string by doing `sprint(PGFPlotsX.print_tex, is, pgfx_fillstyle(opt, i))`

Reply:
I did not know about this, cool and thanks for educating me!